src/main/java/wolox/training/models/Book.java

```diff
@@ -0,0 +1,136 @@
+package wolox.training.models;
+
+import com.sun.istack.NotNull;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @Column(nullable = false)
+    @NotNull
+    private String genre;
+
+    @Column(nullable = false)
+    @NotNull
+    private String author;
+
+    @Column(nullable = false)
+    @NotNull
+    private String image;
+
+    @Column(nullable = false)
+    @NotNull
+    private String title;
+
+    @Column(nullable = false)
+    @NotNull
+    private String subTitle;
+
+    @Column(nullable = false)
+    @NotNull
+    private String publisher;
+
+    @Column(nullable = false)
+    @NotNull
+    private String year;
+
+    @Column(nullable = false)
+    @NotNull
+    private Integer pages;
+
+    @Column(nullable = false)
+    @NotNull
+    private String isbn;
+
+    public Book() {
+        //Constructor for JPA
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
```

Comment:
This setter should not be included because the id is autogenerated and you don't want it to be modified from outside 🙂

Reply:
yes, you are absolutely right